scheduler/reconcile.go

```diff
@@ -796,3 +830,71 @@ func (a *allocReconciler) computeUpdates(group *structs.TaskGroup, untainted all
 
 	return
 }
+func (a *allocReconciler) handleDelayedReschedules(rescheduleLater []*delayedRescheduleInfo, all allocSet, tgName string) allocSet {
+	// Sort by time
```

Comment:
Could you simplify this by:

1. Return early if the rescheduleLater list is empty
2. After time sorting, create an initial eval
3. In your loop add to batch based on the eval.WaitUntil time and when exceeded, create the new eval.

I think that removes the fence post problem

Reply:
@dadgar I dropped in to check out the Nomad project and have been following the PRs for a bit. I don't have context on the `fence post problem` and I did not have confidence I found the right results when Googling.

Can you elaborate or link me to a resource for my education? 

Respectfully,

Aaron

Reply:
I believe Alex is referring to the off by one case I handle below 

https://en.wikipedia.org/wiki/Off-by-one_error

Reply:
@aarongreenlee That wiki page captures it!